applications/nrf5340_audio/src/bluetooth/bt_rendering_and_capture/volume/bt_vol_ctlr_internal.c

```diff
@@ -0,0 +1,262 @@
+		if (ret) {
+			LOG_DBG("Failed to sync volume to remote device %d, err = "
+				"%d",
+				i, ret);
+		}
+	}
+}
+
+/**
+ * @brief	Callback handler for the VCS controller flags.
+ *
+ * @note	This callback handler will be triggered if VCS flags changed.
+ */
+static void vcs_flags_ctlr_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t flags)
+{
+	if (err) {
+		LOG_ERR("VCS flag callback error: %d", err);
+	} else {
+		LOG_DBG("Volume flags = 0x%01X", flags);
+	}
+}
+
+/**
+ * @brief	Callback handler for the finished VCS discovery.
+ *
+ * @note	This callback handler will be triggered when the VCS discovery has finished.
+ */
+static void vcs_discover_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t vocs_count,
+				    uint8_t aics_count)
+{
+	if (err) {
+		LOG_WRN("VCS discover finished callback error: %d", err);
+	} else {
+		LOG_INF("VCS discover finished");
+	}
+}
+
+int bt_vol_ctlr_set(uint8_t volume)
+{
+	int ret;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] != NULL) {
+			ret = bt_vcp_vol_ctlr_set_vol(vcs_client_peer[i], volume);
+			if (ret) {
+				LOG_WRN("Failed to set volume for remote channel %d, ret = "
+					"%d",
+					i, ret);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int bt_vol_ctlr_up(void)
+{
+	int ret;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] != NULL) {
+			ret = bt_vcp_vol_ctlr_unmute_vol_up(vcs_client_peer[i]);
+			if (ret) {
+				LOG_WRN("Failed to volume up for remote channel %d, ret = "
+					"%d",
+					i, ret);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int bt_vol_ctlr_down(void)
+{
+	int ret;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] != NULL) {
+			ret = bt_vcp_vol_ctlr_unmute_vol_down(vcs_client_peer[i]);
+			if (ret) {
+				LOG_WRN("Failed to volume down for remote channel %d, ret "
+					"= %d",
+					i, ret);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int bt_vol_ctlr_mute(void)
+{
+	int ret;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] != NULL) {
+			ret = bt_vcp_vol_ctlr_mute(vcs_client_peer[i]);
+			if (ret) {
+				LOG_WRN("Failed to mute for remote channel %d, ret "
+					"= %d",
+					i, ret);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int bt_vol_ctlr_unmute(void)
+{
+
+	int ret;
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (vcs_client_peer[i] != NULL) {
+			ret = bt_vcp_vol_ctlr_unmute(vcs_client_peer[i]);
+			if (ret) {
+				LOG_WRN("Failed to unmute for remote channel %d, "
+					"ret = %d",
+					i, ret);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int bt_vol_ctlr_discover(struct bt_conn *conn)
+{
+
+	int ret, index;
+
+	if (vcs_client_peer_exists(conn)) {
+		return -EAGAIN;
+	}
+
+	index = vcs_client_peer_index_free_get();
+	if (index < 0) {
+		return index;
+	}
+
+	ret = bt_vcp_vol_ctlr_discover(conn, &vcs_client_peer[index]);
+	return ret;
+}
+
+int bt_vol_ctlr_init(void)
+{
+	if (!IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
```

Comment:
This file is not compiled unless `CONFIG_BT_VCP_VOL_CTLR` is enabled, so we don't need this check anymore. I guess someone could manually add this c-file, but then they are on their own...